lab_5_scrapper/scrapper.py

```diff
@@ -120,46 +258,226 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Finds meta information of article
         """
-        pass
+        author_elem = article_soup.find('a', {'class': 'page-main__publish-author'})
+        self.article.author = [elem.get_text(strip=True) for elem in author_elem] \
+            if author_elem else ["NOT FOUND"]
+
+        date_elem = article_soup.find('a', {'class': 'page-main__publish-date'})
+        date_str = date_elem.get_text(strip=True) if date_elem else "NOT FOUND"
+        self.article.date = self.unify_date_format(date_str)
+
+        topic_elem = article_soup.find_all('a', {'class': 'panel-group__title'})[1]
+        self.article.topics = topic_elem.get_text(strip=True) if topic_elem else "NOT FOUND"
+
+        title_elem = article_soup.find('h1', {'class': 'page-main__head'})
+        self.article.title = title_elem.get_text(strip=True) if title_elem else "NOT FOUND"
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        months_dict = {
+            "января": "January",
+            "февраля": "February",
+            "марта": "March",
+            "апреля": "April",
+            "мая": "May",
+            "июня": "June",
+            "июля": "July",
+            "августа": "August",
+            "сентября": "September",
+            "октября": "October",
+            "ноября": "November",
+            "декабря": "December"
+        }
+        date_str = date_str.replace(date_str.split()[1], months_dict[date_str.split()[1]])
+        return datetime.datetime.strptime(date_str, '%d %B %Y - %H:%M')
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = make_request(self.full_url, self.config)
+        article_soup = BeautifulSoup(response.text, 'lxml')
+        self._fill_article_with_text(article_soup)
+        self._fill_article_with_meta_information(article_soup)
+        return self.article
 
 
-def prepare_environment(base_path: Union[Path, str]) -> None:
+def prepare_environment(base_path: Union[Path, str], recursion: bool = False) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        if not recursion:
+            shutil.rmtree(base_path)
+            base_path.mkdir(parents=True)
+    else:
+        base_path.mkdir(parents=True)
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler implementation
+    """
+
+    def __init__(self, config: Config):
+        super().__init__(config)
+        self.crawler_data_path = Path(__file__).parent / 'crawler_data.json'
+        self.start_url = config.get_seed_urls()[0]
+        self.num_visited_urls = 0
+        self.last_file_index = 0
+        self.visited_urls = []
+        self.urls = []
+
+    def _handle_crawler_data(self, mode: str) -> None:
```

Comment:
too encapsulated. do not be afraid to create separate methods for reading and writing - those two are absolutely different things